### PR TITLE
fix: donation dashboard FE/BE changes

### PR DESCRIPTION
**Pull request by Shehryar21**
## Notion ticket link

[Show Donation Cards in Ascending Order by StartTime](https://www.notion.so/uwblueprintexecs/Show-Donation-Cards-in-Ascending-Order-by-StartTime-a28f510aee8f46a089a6cf8fc1646e23)
[Show upcoming donations for next 2 weeks in dashboard and add state when no donations scheduled](https://www.notion.so/uwblueprintexecs/Show-upcoming-donations-for-next-2-weeks-in-dashboard-and-add-state-when-no-donations-scheduled-087550e5ebf64c5bb68ba8bfc47a96f1)

## Implementation description
* Change backend scheduling get endpoint to return donations sorted by start time.
* For get by donorId endpoint, added a weekLimit param that if its not passed in, all donations are returned but if it is, it returns the donations scheduled for those many upcoming weeks

## Steps to test
1. Verify that donations in dashboard are showing in sorted order by start time
2. Verify that donations are shown for upcoming 2 weeks

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/Dashboard/index.tsx [hunk 1]
```diff
@@ -62,17 +68,24 @@ const Dashboard = (): JSX.Element => {
         Upcoming Dropoffs
       </Text>
       <Text pt="0.8rem" textStyle="mobileBody" mb="1.5rem">
-        View all of the upcoming donations that you have scheduled{" "}
+        View all of the upcoming donations that you have scheduled for the next
+        2 weeks
```
Comment (hanlinc27) [inline frontend/src/components/pages/Dashboard/index.tsx:72]:
```suggestion
        View all of the upcoming donations that you have scheduled for the next
        two weeks
```

File: frontend/src/components/pages/Dashboard/index.tsx [hunk 2]
```diff
@@ -22,13 +24,17 @@ const Dashboard = (): JSX.Element => {
   };
 
   React.useEffect(() => {
+    if (!authenticatedUser) {
+      return;
+    }
```
Comment (hanlinc27) [inline frontend/src/components/pages/Dashboard/index.tsx:29]:
```suggestion
    if (!authenticatedUser) {
      return <Redirect to={LOGIN_PAGE}/>;
    }
```

File: frontend/src/components/pages/Dashboard/index.tsx [hunk 3]
```diff
@@ -9,6 +9,8 @@ import AuthContext from "../../../contexts/AuthContext";
 import { Schedule } from "../../../types/SchedulingTypes";
 import DropoffCard from "./components/DropoffCard";
 
+const upcomingWeekLimit = "2";
```
Comment (hanlinc27) [inline frontend/src/components/pages/Dashboard/index.tsx:12]:
```suggestion
const UPCOMING_WEEK_LIMIT = "2";
```
I also think we should move this constant value into a file in `/constants` but up to you

File: backend/typescript/services/implementations/schedulingService.ts [hunk 4]
```diff
@@ -69,12 +70,32 @@ class SchedulingService implements ISchedulingService {
...
-      });
+      if (weekLimit) {
+        const currentStartDate = new Date();
+        const nextDate = new Date();
+        nextDate.setDate(nextDate.getDate() + weekLimit * 7);
+        schedulings = await Scheduling.findAll({
+          where: {
+            donor_id: Number(donorId),
+            start_time: {
+              [Op.between]: [currentStartDate, nextDate] as any,
```
Comment (hanlinc27) [inline backend/typescript/services/implementations/schedulingService.ts:86]:
```suggestion
              [Op.between]: [currentStartDate, nextDate],
```
don't think you need that

File: frontend/src/APIClients/SchedulingAPIClient.ts [hunk 5]
```diff
@@ -25,14 +25,17 @@ const getScheduleById = async (scheduleId: string): Promise<Schedule> => {
...
-      {
-        headers: { Authorization: BEARER_TOKEN },
-      },
-    );
+    const url = weekLimit
+      ? `/scheduling/?donorId=${donorId}&weekLimit=${weekLimit}`
+      : `/scheduling/?donorId=${donorId}`;
+    const { data } = await baseAPIClient.get(url, {
+      headers: { Authorization: BEARER_TOKEN },
+    });
```
Comment (hanlinc27) [inline frontend/src/APIClients/SchedulingAPIClient.ts:38]: what about making `weekLimit` a mandatory parameter, but passing in `0` if there's no limit. This removes the ternary conditions in this file and `schedulingRoutes.ts` and I think is overall a bit cleaner.

Inside of the service we can just keep the same condition but check for `weekLimit > 0`

**Review by hanlinc27 — CHANGES_REQUESTED** [review]

File: frontend/src/components/pages/Dashboard/index.tsx (hunk 2 above)
Comment (Shehryar21) [reply frontend/src/components/pages/Dashboard/index.tsx:29]: this is inside useEffect function, we cannot return `<Redirect/>`

**Pull request merged**